lock-api/src/main/java/com/palantir/lock/StringLockDescriptor.java

```diff
@@ -27,15 +27,16 @@
  *
  * @author jtamer
  */
-public class StringLockDescriptor {
+public final class StringLockDescriptor {
 
     private StringLockDescriptor() {
         // cannot instantiate
     }
 
     /** Returns a {@code LockDescriptor} instance for the given lock ID. */
+    @SuppressWarnings("checkstyle:RegexpSinglelineJava") // StandardCharsets only in JDK 1.7+
```

Comment:
Is there anyway to suppress the specific rule about Charsets rather than all rules implemented using `RegexpSinglelineJava`?  We might miss other breakages here if we suppress them all.